pkg/reporting/preflight.go

```diff
@@ -68,30 +69,67 @@ func SendPreflightInfo(appID string, sequence int, isSkipPreflights bool, isUpda
 
 	clusterID := downstreams[0].ClusterID
 
-	if isSkipPreflights || isUpdate {
-		// at this point current version status does not exist so it's
-		// neccessary to create thread to get it after version is deployed
+	go func() {
+		appStatus := appstatustypes.StateMissing
+		for start := time.Now(); time.Since(start) < 20*time.Minute; {
```

Comment:
is this now a tight loop? it used to poll every 20 seconds

https://github.com/replicatedhq/kots/pull/1712/files#diff-706f7e57e6f7361d82c9a8f1b0c3ddb08c18e88d3ee384cf20964b0ca86254c2L77

Reply:
this didn't use to run every 20 seconds. it was just a one time hard 20 seconds wait.

Reply:
and i think tight loops are fine here since the code will have to wait for the app status anyways before reporting, right?

Reply:
and it breaks once the app becomes ready here https://github.com/replicatedhq/kots/pull/1712/files#diff-706f7e57e6f7361d82c9a8f1b0c3ddb08c18e88d3ee384cf20964b0ca86254c2R81

Reply:
But wont it starve everything else of cpu? or is there a sleep somewhere im not seeing?

Reply:
Also its querying the db in GetAppStatus

Reply:
oh, good call, i didn't notice there's no sleep.

Reply:
resolved, added sleep